src/NHibernate/Engine/ActionQueue.cs

```diff
@@ -151,7 +170,6 @@ private void RegisterCleanupActions(IExecutable executable)
 			{
 				string[] spaces = executable.PropertySpaces;
 				afterTransactionProcesses.AddSpacesToInvalidate(spaces);
```

Comment:
I think it is better to utilize `afterTransactionProcesses` instead of creating an additional hashset.